src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,157 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T value) {
+            this.value = value;
+            this.prev = null;
+            this.next = null;
+        }
+    }
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (head == null && tail == null) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = tail.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            checkIndex(index);
+            Node<T> newNode = new Node<>(value);
+            Node<T> currentNode = getNodeByIndex(index);
+            if (isBetween(currentNode) || isTail(currentNode)) {
+                currentNode.prev.next = newNode;
+                newNode.next = currentNode;
+                newNode.prev = currentNode.prev;
+                currentNode.prev = newNode;
+            } else if (isHead(currentNode)) {
+                head = newNode;
+                head.next = currentNode;
+                currentNode.prev = head;
+            }
+            size++;
+        }
```

Comment:
1. It's better to avoid additional tabulations where possible to make code more readable
2. Conditions seems over-designed. As we checking it's same object we may use == in such case
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkIndex(index);
        Node<T> newNode = new Node<>(value);
        Node<T> currentNode = getNodeByIndex(index);
        if (currentNode == head) {
            head = newNode;
            head.next = currentNode;
            currentNode.prev = head;
        } else {
            currentNode.prev.next = newNode;
            newNode.next = currentNode;
            newNode.prev = currentNode.prev;
            currentNode.prev = newNode;
        }
        size++;
```